### DISCUSSION TITLE
Substituting values in the middle of a ConfigMap

### DISCUSSION THREAD

**Discussion by pinkfloydx33** [kubernetes-sigs/kustomize#5153] [Q&A]
We have some configmaps where the keys represent structured config files (yaml, json, toml) for various components of our infrastructure. For the most part, many of these files are the same across all usages. However there are instances where we need to replace a single token in the _middle_ of one of these values. For example replacing the word "dev" with "prod" in some banner text or "warn" with "trace" for a log level.

It looks like the deprecated `vars` _could_ have solved this; `replacements` wouldn't work as the value is buried in the middle of a file which itself is embedded in a configmap. 

At the moment our only course of action is to maintain multiple copies of these configmap/files. Is there any option that I'm overlooking that would allow us to easily perform substitution inside these values?

Note: `envsubst` isn't an option. We are using kustomize inside of AKS Gitops/FluxCD which reads files directly out of git repository. Flux has a `postBuild.substitutions` that _would_ work under normal circumstances except for the fact that the Azure managed Flux doesn't allow setting those values.

**Comment by nilsherzig** [comment]
hi can you remember if you found a solution for this problem? Facing the same issue over here :(

**Reply by pinkfloydx33** [reply]
Nope. Sorry.

**Reply by sda399** [reply]
you can create a krm function for this (as a template). but then Flux will be the limiter, as they prefer you go the oci way per their github issues.
